compiler/src/iree/compiler/Codegen/Common/DecomposePackUnPackOps.cpp

```diff
@@ -243,11 +264,106 @@ void DecomposePackUnPackOpsPass::runOnOperation() {
   }
 }
 
-std::unique_ptr<InterfacePass<mlir::FunctionOpInterface>>
+static std::unique_ptr<InterfacePass<mlir::FunctionOpInterface>>
 createDecomposePackUnPackOpsPass(bool tileOuterToOne, bool useOnlyReshapes,
                                  std::optional<PackUnPackControlFn> controlFn) {
   return std::make_unique<DecomposePackUnPackOpsPass>(
       tileOuterToOne, useOnlyReshapes, controlFn);
 }
```

Comment:
This is here to pass the controlFn to the pass. It is not public because we don't want to be passing control functions in the codegen pipelines. It makes it easier to create repros because you can run it in the command line with the desired control function this way.